shell/platform/fuchsia/flutter/pointer_delegate.cc

```diff
@@ -318,6 +320,21 @@ void InsertIntoBuffer(
 }
 }  // namespace
 
+PointerDelegate::PointerDelegate(
+    fuchsia::ui::pointer::TouchSourceHandle touch_source,
+    fuchsia::ui::pointer::MouseSourceHandle mouse_source)
+    : touch_source_(touch_source.Bind()), mouse_source_(mouse_source.Bind()) {
+  touch_source_.set_error_handler([](zx_status_t status) {
```

Comment:
Why do we do `if (mouse_source_)` but not `if (touch_source_)`? Should we add a comment here explaining, and maybe `FML_CHECK(touch_source_)` before setting the error handler?

Reply:
I can leave a comment. On today's $SHIPPING_PRODUCT, GFX runs in touch-only mode, so we plug in TouchSource but not MouseSource (in engine.cc). This is meant to reduce the risk of "something going awry". In contrast, Flatland needs both on Workstation, so we have both plugged in. 

We might need a better story for being able to selectively turn on/off input APIs on specific products. Would you like a bug filed for that?

Reply:
I'm okay without a bug for that because I'm not sure what the immediate action would be and who the prospective client would be. Maybe we should wait for a client before making the feature request since there's enough stuff on the Input team's plate anyways.

That being said would it make our code more future-proof to wrap this call to `touch_source_.set_error_handler` in an `if (touch_source_)`? If not, we don't have to do it but it does seem somewhat unsafe as currently written (I doubt TouchSource will always be available on every product?), so I'd like either the if statement or the `FML_CHECK` (preferably the if statement since it's more flexible).

Reply:
That seems quite reasonable. The symmetry is pleasing too. I'll upload a patch.